doc/developer/entrypoints.rst

```diff
@@ -0,0 +1,82 @@
+Entry Points
+============
+
+**Entry Points** are extensible routines that you can choose
+when you invoke your application. The :py:mod:`built-in entry points <armi.cli>` 
+offer basic functionality, like :py:class:`running a case <armi.cli.run.RunEntryPoint>`
+or :py:class:`opening up the GUI <armi.cli.gridGui.GridGuiEntryPoint>`, but
+the real joy of an application comes when you add your own project-specific 
+entry points that do the actions that you commonly need done.
+
+To make a new EntryPoint, first make a new module and subclass 
+:py:class:`~armi.cli.entryPoint.EntryPoint`. Set the 
+class attributes as follows:
+
+``name``
+    What the user types on the CLI to invoke this entry point. 
+
+``settingsArgument``
+
+    * ``"required"`` if a settings input file must be provided,
+    * ``"optional"`` if it may be provided but not required, 
+    * ``None`` if no settings input is allowed
+
+
+.. tip:: ARMI apps often collect EntryPoints in a ``cli/`` directory (Command Line Interface)
+
+Next, implement the :py:meth:`~armi.cli.entryPoint.EntryPoint.addOptions` method. Here you can both:
+
+* turn various App Settings into command-line arguments with :py:meth:`~armi.cli.entryPoint.EntryPoint.createOptionFromSetting`
+* add arbitrary other command-line arguments using the standard :py:mod:`python:argparse` library.
+
+The values of the non-setting arguments will become attributes in ``self.args`` for later use.
```

Comment:
Now that intersphinx is set up, we might want to link the relevant `argparse` docs

Reply:
that's what this does, and that's why I added intersphinx.

Reply:
oh im a dummy, its right there. i didnt see the link 🤦‍♂️